File: app/src/main/java/com/brian_bui/true_clothes/home/ItemDetailScreen.kt
package com.brian_bui.true_clothes.home

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.heightIn
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.statusBarsPadding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.platform.LocalUriHandler
import androidx.compose.ui.semantics.Role
import androidx.compose.ui.semantics.contentDescription
import androidx.compose.ui.semantics.role
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.text.style.TextDecoration
import androidx.compose.ui.tooling.preview.Preview
import com.brian_bui.true_clothes.shared.media.BitmapSourceImage
import com.brian_bui.true_clothes.shared.theme.DefaultBodyFontFamily
import com.brian_bui.true_clothes.shared.theme.DefaultHeaderFontFamily
import com.brian_bui.true_clothes.shared.theme.DefaultScreenBackground
import com.brian_bui.true_clothes.shared.theme.LocalResponsiveScale
import com.brian_bui.true_clothes.shared.theme.TrueclothesTheme

@Composable
fun ItemDetailScreen(
    model: ItemDetailUi,
    onBack: () -> Unit,
) {
    val scale = LocalResponsiveScale.current
    val uriHandler = LocalUriHandler.current
    val horizontal = scale.scaleDp(28f)
    val heroHeight = scale.scaleDp(486f)
    val rows = measurementRowsFor(model.category, model.measurements)
    val scrimHeight = scale.scaleDp(140f)

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(DefaultScreenBackground)
            .verticalScroll(rememberScrollState()),
    ) {
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .height(heroHeight),
        ) {
            Box(
                modifier = Modifier
                    .fillMaxSize()
                    .background(Color(0xFFDCDCDC)),
            )
            BitmapSourceImage(
                imageSource = model.imageSource,
                contentDescription = model.fullName,
                modifier = Modifier.fillMaxSize(),
                contentScale = ContentScale.Crop,
            )
            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .height(scrimHeight)
                    .align(Alignment.TopCenter)
                    .background(
                        Brush.verticalGradient(
                            colors = listOf(
                                Color.Black.copy(alpha = 0.55f),
                                Color.Transparent,
                            ),
                        ),
                    ),
            )
            Row(
                modifier = Modifier
                    .align(Alignment.TopStart)
                    .statusBarsPadding()
                    .padding(
                        start = scale.scaleDp(8f),
                        top = scale.scaleDp(8f),
                    ),
                verticalAlignment = Alignment.CenterVertically,
            ) {
                TextButton(
                    onClick = onBack,
                    colors = ButtonDefaults.textButtonColors(contentColor = Color.White),
                ) {
                    Text(
                        text = "Back",
                        style = MaterialTheme.typography.bodyMedium.copy(
                            fontFamily = DefaultBodyFontFamily,
                            fontWeight = FontWeight.Bold,
                        ),
                    )
                }
            }
        }

        Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = horizontal)
                    .padding(top = scale.scaleDp(24f)),
            ) {
                Text(
                    text = model.fullName,
                    style = MaterialTheme.typography.titleLarge.copy(
                        fontFamily = DefaultHeaderFontFamily,
                        fontWeight = FontWeight.Bold,
                        fontSize = scale.scaleSp(22f),
                        lineHeight = scale.scaleSp(34f),
                    ),
                    color = Color(0xFF000000),
                )

                Spacer(modifier = Modifier.height(scale.scaleDp(12f)))

                Text(
                    text = model.color,
                    style = MaterialTheme.typography.bodyMedium.copy(
                        fontFamily = DefaultBodyFontFamily,
                        fontWeight = FontWeight.Light,
                        fontSize = scale.scaleSp(20f),
                    ),
                    color = Color(0xFF000000).copy(alpha = 0.5f),
                    modifier = Modifier.semantics {
                        contentDescription = "Color: ${model.color}"
                    },
                )

                Spacer(modifier = Modifier.height(scale.scaleDp(8f)))

                Text(
                    text = "${model.brand}",
                    style = MaterialTheme.typography.bodyMedium.copy(
                        fontFamily = DefaultBodyFontFamily,
                        fontWeight = FontWeight.Medium,
                        fontSize = scale.scaleSp(15f),
                    ),
                    color = Color(0xFF000000),
                )

                Spacer(modifier = Modifier.height(scale.scaleDp(20f)))

                val url = model.productUrl
                if (!url.isNullOrBlank()) {
                    val host = runCatching { android.net.Uri.parse(url).host }.getOrNull().orEmpty()
                    val linkSemantics = if (host.isNotEmpty()) {
                        "View product link. Opens $host"
                    } else {
                        "View product link"
                    }
                    Text(
                        text = "View product link",
                        style = TextStyle(
                            fontFamily = DefaultBodyFontFamily,
                            fontWeight = FontWeight.SemiBold,
                            fontSize = scale.scaleSp(16f),
                            textDecoration = TextDecoration.Underline,
                        ),
                        color = Color(0xFF000000),
                        modifier = Modifier
                            .heightIn(min = scale.scaleDp(48f))
                            .semantics {
                                role = Role.Button
                                contentDescription = linkSemantics
                            }
                            .clickable {
                                runCatching { uriHandler.openUri(url) }
                            }
                            .padding(vertical = scale.scaleDp(8f)),
                    )
                }

                Spacer(modifier = Modifier.height(scale.scaleDp(28f)))

                Text(
                    text = "Measurements",
                    style = MaterialTheme.typography.titleMedium.copy(
                        fontFamily = DefaultBodyFontFamily,
                        fontWeight = FontWeight.SemiBold,
                        fontSize = scale.scaleSp(16f),
                    ),
                    color = Color(0xFF000000),
                )

                Spacer(modifier = Modifier.height(scale.scaleDp(8f)))

                Text(
                    text = model.measurementUnitNote,
                    style = MaterialTheme.typography.bodySmall.copy(
                        fontFamily = DefaultBodyFontFamily,
                        fontWeight = FontWeight.Normal,
                        fontSize = scale.scaleSp(13f),
                    ),
                    color = Color(0xFF000000).copy(alpha = 0.55f),
                )

                Spacer(modifier = Modifier.height(scale.scaleDp(12f)))

                if (rows.isEmpty()) {
                    Text(
                        text = "No measurements added",
                        style = MaterialTheme.typography.bodyMedium.copy(
                            fontFamily = DefaultBodyFontFamily,
                            fontSize = scale.scaleSp(14f),
                        ),
                        color = Color(0xFF000000).copy(alpha = 0.55f),
                    )
                } else {
                    rows.forEachIndexed { index, (label, value) ->
                        if (index > 0) {
                            Box(
                                modifier = Modifier
                                    .fillMaxWidth()
                                    .height(scale.scaleDp(0.3f))
                                    .background(Color.Black.copy(alpha = 0.15f)),
                            )
                        }
                        Row(
                            modifier = Modifier
                                .fillMaxWidth()
                                .heightIn(min = scale.scaleDp(48f))
                                .padding(vertical = scale.scaleDp(12f)),
                            verticalAlignment = Alignment.CenterVertically,
                        ) {
                            Text(
                                text = label,
                                style = MaterialTheme.typography.bodyMedium.copy(
                                    fontFamily = DefaultBodyFontFamily,
                                    fontWeight = FontWeight.Normal,
                                    fontSize = scale.scaleSp(14f),
                                ),
                                color = Color(0xFF000000).copy(alpha = 0.55f),
                                modifier = Modifier.weight(1f),
                            )
                            Text(
                                text = value,
                                style = MaterialTheme.typography.bodyMedium.copy(
                                    fontFamily = DefaultBodyFontFamily,
                                    fontWeight = FontWeight.Medium,
                                    fontSize = scale.scaleSp(14f),
                                ),
                                color = Color(0xFF000000),
                                textAlign = TextAlign.End,
                                modifier = Modifier.padding(start = scale.scaleDp(12f)),
                            )
                        }
                    }
                }

                Spacer(modifier = Modifier.height(scale.scaleDp(32f)))
            }
    }
}

@Preview(showBackground = true)
@Composable
private fun ItemDetailScreenPreview() {
    val model = ItemDetailUi(
        itemId = "preview-shirt",
        imageSource = "asset:images/shirt.png",
        fullName = "Poplin spread-collar dress shirt",
        color = "BLACK",
        brand = "Brooks Brothers",
        productUrl = "https://example.com/product/shirt",
        category = ItemCategory.Top,
        measurements = mapOf(
            "chest" to "98 cm",
            "sleeves" to "64 cm",
            "shoulder" to "45 cm",
            "bicep" to "34 cm",
            "length" to "78 cm",
        ),
    )
    TrueclothesTheme {
        ItemDetailScreen(model = model, onBack = {})
    }
}

File: true_clothes_app/lib/features/home/item_detail_screen.dart
import 'package:flutter/material.dart';
import 'package:flutter/widget_previews.dart';
import 'package:url_launcher/url_launcher.dart';

import '../../core/responsive.dart';
import '../../theme/app_colors.dart';
import '../../theme/app_fonts.dart';
import '../../theme/app_theme.dart';
import '../../theme/widget_preview_theme.dart';
import '../../widgets/outfit_composition_card_placeholder.dart';
import 'home_outfit_models.dart';

// ---------------------------------------------------------------------------
// Measurement field order per category — mirrors Kotlin `measurementRowsFor`.
// ---------------------------------------------------------------------------

const _categoryFields = <ItemCategory, List<({String key, String label})>>{
  ItemCategory.top: [
    (key: 'chest', label: 'Chest'),
    (key: 'sleeves', label: 'Sleeves'),
    (key: 'shoulder', label: 'Shoulder'),
    (key: 'bicep', label: 'Bicep'),
    (key: 'length', label: 'Length'),
  ],
  ItemCategory.bottom: [
    (key: 'waist', label: 'Waist'),
    (key: 'inseam', label: 'Inseam'),
    (key: 'thigh', label: 'Thigh'),
    (key: 'knee', label: 'Knee'),
    (key: 'leg_opening', label: 'Leg opening'),
    (key: 'rise', label: 'Rise'),
  ],
  ItemCategory.outwear: [
    (key: 'chest', label: 'Chest'),
    (key: 'shoulder', label: 'Shoulder'),
    (key: 'sleeves', label: 'Sleeves'),
    (key: 'length', label: 'Length'),
    (key: 'bicep', label: 'Bicep'),
  ],
  ItemCategory.shoes: [
    (key: 'size_us', label: 'Size (US)'),
    (key: 'size_eu', label: 'Size (EU)'),
    (key: 'foot_length', label: 'Foot length'),
    (key: 'width', label: 'Width'),
    (key: 'instep', label: 'Instep'),
  ],
  ItemCategory.accessory: [
    (key: 'length', label: 'Length'),
    (key: 'width', label: 'Width'),
    (key: 'height', label: 'Height'),
    (key: 'circumference', label: 'Circumference'),
    (key: 'strap_drop', label: 'Strap drop'),
  ],
};

List<(String label, String value)> _measurementRows(
  ItemCategory category,
  Map<String, String> measurements,
) {
  final fields = _categoryFields[category] ?? [];
  return [
    for (final f in fields)
      if (measurements[f.key] case final v? when v.isNotEmpty) (f.label, v),
  ];
}

String _measurementUnitNote(Map<String, String> measurements) {
  if (measurements.isEmpty) return '';
  final allCm = measurements.values.every((v) => v.trim().endsWith('cm'));
  return allCm ? 'All values in cm' : '';
}

// ---------------------------------------------------------------------------
// Screen
// ---------------------------------------------------------------------------

/// Matches Kotlin `ItemDetailScreen.kt` + `app/design/screen/home/item/design.md`.
class ItemDetailScreen extends StatelessWidget {
  const ItemDetailScreen({
    super.key,
    required this.item,
    required this.onBack,
  });

  final OutfitItemInfo item;
  final VoidCallback onBack;

  @override
  Widget build(BuildContext context) {
    final horizontal = scaleDp(context, 28);
    final heroHeight = scaleDp(context, 486);
    final scrimHeight = scaleDp(context, 140);
    final statusBarTop = MediaQuery.paddingOf(context).top;
    final rows = _measurementRows(item.category, item.measurements);
    final unitNote = _measurementUnitNote(item.measurements);
    final url = item.productUrl;

    return Scaffold(
      backgroundColor: AppColors.mainBackground,
      // No AppBar — body starts at y=0 so the hero is flush to screen top,
      // matching Kotlin's Column(fillMaxSize) with no top window inset offset.
      body: SingleChildScrollView(
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.stretch,
          children: [
            // ── Hero banner ──────────────────────────────────────────────
            SizedBox(
              width: double.infinity,
              height: heroHeight,
              child: Stack(
                fit: StackFit.expand,
                children: [
                  // Placeholder background (#DCDCDC) shown before/on error.
                  const ColoredBox(color: Color(0xFFDCDCDC)),

                  // Product image — full bleed, ContentScale.Crop.
                  if (item.imageSource != null && item.imageSource!.isNotEmpty)
                    Image.asset(
                      item.imageSource!,
                      package: kPreviewImagePackage,
                      fit: BoxFit.cover,
                      width: double.infinity,
                      height: heroHeight,
                      errorBuilder: (_, _, _) => const SizedBox.shrink(),
                    ),

                  // Gradient scrim: top 140dp, black 55% → transparent.
                  Positioned(
                    top: 0,
                    left: 0,
                    right: 0,
                    height: scrimHeight,
                    child: const DecoratedBox(
                      decoration: BoxDecoration(
                        gradient: LinearGradient(
                          begin: Alignment.topCenter,
                          end: Alignment.bottomCenter,
                          colors: [
                            Color(0x8C000000), // black ~55%
                            Colors.transparent,
                          ],
                        ),
                      ),
                    ),
                  ),

                  // Back button — TopStart, below status bar + 8dp, start + 8dp.
                  Positioned(
                    top: statusBarTop + scaleDp(context, 8),
                    left: scaleDp(context, 8),
                    child: TextButton(
                      onPressed: onBack,
                      style: TextButton.styleFrom(
                        padding: EdgeInsets.zero,
                        minimumSize: Size.zero,
                        tapTargetSize: MaterialTapTargetSize.shrinkWrap,
                        foregroundColor: Colors.white,
                      ),
                      child: Text(
                        'Back',
                        style: AppFonts.poppins(
                          context,
                          fontSize: scaleSp(context, 14),
                          fontWeight: FontWeight.w700,
                          color: Colors.white,
                        ),
                      ),
                    ),
                  ),
                ],
              ),
            ),

            // ── Content scroll ───────────────────────────────────────────
            Padding(
              padding: EdgeInsets.only(
                left: horizontal,
                right: horizontal,
                top: scaleDp(context, 24),
              ),
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  // Full product name — Playfair Display Bold 22sp / lh 34sp.
                  Text(
                    item.fullName,
                    style: AppFonts.playfairDisplay(
                      context,
                      fontSize: scaleSp(context, 22),
                      fontWeight: FontWeight.bold,
                      height: scaleSp(context, 34) / scaleSp(context, 22),
                      color: Colors.black,
                    ),
                  ),

                  SizedBox(height: scaleDp(context, 12)),

                  // Color — value only; accessibility label includes "Color:" prefix.
                  Semantics(
                    label: 'Color: ${item.itemColor}',
                    child: ExcludeSemantics(
                      child: Text(
                        item.itemColor,
                        style: AppFonts.poppins(
                          context,
                          fontSize: scaleSp(context, 20),
                          fontWeight: FontWeight.w300,
                          color: Colors.black.withValues(alpha: 0.5),
                        ),
                      ),
                    ),
                  ),

                  SizedBox(height: scaleDp(context, 8)),

                  // Brand name — Poppins Medium 15sp (no "Brand:" prefix, matches Kotlin).
                  Text(
                    item.brand,
                    style: AppFonts.poppins(
                      context,
                      fontSize: scaleSp(context, 15),
                      fontWeight: FontWeight.w500,
                      color: Colors.black,
                    ),
                  ),

                  SizedBox(height: scaleDp(context, 20)),

                  // Product link — only when URL is present.
                  if (url != null && url.isNotEmpty)
                    _ProductLink(url: url, context: context),

                  SizedBox(height: scaleDp(context, 28)),

                  // Measurements section title.
                  Text(
                    'Measurements',
                    style: AppFonts.poppins(
                      context,
                      fontSize: scaleSp(context, 16),
                      fontWeight: FontWeight.w600,
                      color: Colors.black,
                    ),
                  ),

                  SizedBox(height: scaleDp(context, 8)),

                  // Unit note line.
                  if (unitNote.isNotEmpty)
                    Text(
                      unitNote,
                      style: AppFonts.poppins(
                        context,
                        fontSize: scaleSp(context, 13),
                        fontWeight: FontWeight.w400,
                        color: Colors.black.withValues(alpha: 0.55),
                      ),
                    ),

                  SizedBox(height: scaleDp(context, 12)),

                  // Measurement rows or empty state.
                  if (rows.isEmpty)
                    Text(
                      'No measurements added',
                      style: AppFonts.poppins(
                        context,
                        fontSize: scaleSp(context, 14),
                        fontWeight: FontWeight.w400,
                        color: Colors.black.withValues(alpha: 0.55),
                      ),
                    )
                  else
                    for (final (index, (label, value)) in rows.indexed) ...[
                      if (index > 0)
                        Container(
                          height: scaleDp(context, 0.3),
                          color: Colors.black.withValues(alpha: 0.15),
                        ),
                      ConstrainedBox(
                        constraints: BoxConstraints(
                          minHeight: scaleDp(context, 48),
                        ),
                        child: Padding(
                          padding: EdgeInsets.symmetric(
                            vertical: scaleDp(context, 12),
                          ),
                          child: Row(
                            crossAxisAlignment: CrossAxisAlignment.center,
                            children: [
                              Expanded(
                                child: Text(
                                  label,
                                  style: AppFonts.poppins(
                                    context,
                                    fontSize: scaleSp(context, 14),
                                    fontWeight: FontWeight.w400,
                                    color: Colors.black.withValues(alpha: 0.55),
                                  ),
                                ),
                              ),
                              Padding(
                                padding: EdgeInsets.only(
                                  left: scaleDp(context, 12),
                                ),
                                child: Text(
                                  value,
                                  textAlign: TextAlign.end,
                                  style: AppFonts.poppins(
                                    context,
                                    fontSize: scaleSp(context, 14),
                                    fontWeight: FontWeight.w500,
                                    color: Colors.black,
                                  ),
                                ),
                              ),
                            ],
                          ),
                        ),
                      ),
                    ],

                  SizedBox(height: scaleDp(context, 32)),
                ],
              ),
            ),
          ],
        ),
      ),
    );
  }
}

// ---------------------------------------------------------------------------
// Product link widget — Poppins SemiBold 16sp, underline, heightIn(min=48).
// ---------------------------------------------------------------------------

class _ProductLink extends StatelessWidget {
  const _ProductLink({required this.url, required this.context});

  final String url;
  final BuildContext context;

  @override
  Widget build(BuildContext _) {
    return Semantics(
      button: true,
      label: _semanticsLabel(url),
      child: InkWell(
        onTap: () => _openUrl(url),
        child: ConstrainedBox(
          constraints: BoxConstraints(minHeight: scaleDp(context, 48)),
          child: Padding(
            padding: EdgeInsets.symmetric(vertical: scaleDp(context, 8)),
            child: Text(
              'View product link',
              style: AppFonts.poppins(
                context,
                fontSize: scaleSp(context, 16),
                fontWeight: FontWeight.w600,
                color: Colors.black,
              ).copyWith(decoration: TextDecoration.underline),
            ),
          ),
        ),
      ),
    );
  }

  String _semanticsLabel(String url) {
    try {
      final host = Uri.parse(url).host;
      if (host.isNotEmpty) return 'View product link. Opens $host';
    } catch (_) {}
    return 'View product link';
  }

  Future<void> _openUrl(String url) async {
    final uri = Uri.tryParse(url);
    if (uri != null) await launchUrl(uri, mode: LaunchMode.externalApplication);
  }
}

// ---------------------------------------------------------------------------
// Widget Previewer
// ---------------------------------------------------------------------------

class ItemDetailScreenPreview extends StatelessWidget {
  const ItemDetailScreenPreview({super.key});

  @override
  Widget build(BuildContext context) {
    final item = demoOutfitDetailPayload().items.first;
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      theme: buildAppTheme(),
      home: ItemDetailScreen(
        item: item,
        onBack: () {},
      ),
    );
  }
}

@Preview(
  name: 'Item detail',
  group: 'Home',
  size: Size(402, 874),
  theme: trueClothesPreviewTheme,
)
Widget itemDetailScreenWidgetPreview() {
  return const ItemDetailScreenPreview();
}
